### DISCUSSION TITLE
Discrepancy between Total Energy / Balance / Sum

### DISCUSSION THREAD

**Discussion by FedgarLurch** [emporia-vue-local/esphome#286] [Q&A]
Hello all! 
Okay, I am trying to gather my thoughts and present a well defined problem, but I might ramble so fair warning! 

I have been messing with the Emporia Vue 2 for a few weeks now. It worked great on its original firmware, minus the lack of local support. (Plus I love ESPhome)

Since getting it installed, I have checked, rechecked and tinkered with the configuration and I have gotten things dialed in pretty well. One minor issue being that my "Total Daily Energy" sensor doesnt seem to be accurate. 

The first row is the measurement from the Utility, so this is the number that I know (or pretty close to know) is accurate. 
The next row is the sensor.total_daily_energy, and for some reason it is plus or minus 20% of what it should be. 
The last row is the sum of all of the individual circuits, its pretty much dead accurate. 

So the question is... what am I missing? 

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/FRANKE~1/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/FRANKE~1/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#467886" vlink="#96607D">

  | 12:00 AM | 1:00 AM | 2:00 AM | 3:00 AM | 4:00 AM | 5:00 AM | 6:00 AM | 7:00 AM | 8:00 AM | 9:00 AM | 10:00 AM | 11:00 AM | 12:00 PM | 1:00 PM | 2:00 PM | 3:00 PM | 4:00 PM | 5:00 PM | 6:00 PM | 7:00 PM | 8:00 PM | 9:00 PM | 10:00 PM | 11:00 PM
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
UTILITY NUMBER | 2.89 | 1.86 | 1.81 | 1.38 | 1.59 | 0.46 | 0.35 | 0.42 | 0.88 | 0.39 | 1.16 | 2.7 | 3.42 | 3.72 | 4.45 | 3.85 | 5.03 | 4.9 | 4.85 | 3.48 | 3.08 | 3.75 | 4.44 | 6.66
sensor.total_daily_energy | 2.578 | 1.645 | 1.487 | 1.174 | 1.462 | 0.441 | 0.442 | 0.435 | 0.801 | 0.446 | 1.002 | 2.248 | 2.864 | 3.15 | 3.883 | 3.479 | 4.944 | 4.893 | 4.859 | 3.384 | 2.878 | 3.494 | 4.049 | 5.976
sensor.sum | 2.905 | 1.817 | 1.847 | 1.365 | 1.582 | 0.439 | 0.336 | 0.394 | 0.859 | 0.362 | 1.141 | 2.643 | 3.434 | 3.752 | 4.49 | 3.803 | 5.066 | 4.951 | 4.891 | 3.496 | 3.019 | 3.77 | 4.348 | 6.692
  |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |  
sensor.1_closet_bathroom_lights_energy | 0.052 | 0.001 | 0.001 | 0.001 | 0 | 0.001 | 0 | 0.008 | 0.012 | 0.012 | 0.012 | 0.012 | 0.012 | 0.012 | 0.012 | 0.012 | 0.012 | 0.013 | 0.012 | 0.012 | 0.012 | 0.012 | 0.012 | 0.024
sensor.2_kitchen_dining_area_lights_energy | 0 | 0.001 | 0 | 0 | 0.001 | 0 | 0.001 | 0 | 0 | 0.001 | 0 | 0.001 | 0 | 0 | 0.001 | 0 | 0.001 | 0 | 0 | 0.001 | 0 | 0.001 | 0 | 0.001
sensor.3_stove_energy | 0.002 | 0.002 | 0.002 | 0.002 | 0.003 | 0.002 | 0.002 | 0.002 | 0.002 | 0.002 | 0.002 | 0.002 | 0.002 | 0.002 | 0.002 | 0.002 | 0.002 | 0.002 | 0.001 | 0.002 | 0.002 | 0.002 | 0.002 | 0.002
sensor.4_stove_energy | 0.001 | 0 | 0.001 | 0 | 0.001 | 0 | 0.001 | 0 | 0.001 | 0 | 0.001 | 0.001 | 0 | 0.001 | 0 | 0.001 | 0 | 0.001 | 0 | 0.001 | 0 | 0.001 | 0.001 | 0
sensor.5_kitchen_living_room_lights_energy | 0.034 | 0.011 | 0.011 | 0.012 | 0.011 | 0.011 | 0.012 | 0.011 | 0.01 | 0.011 | 0.011 | 0.01 | 0.011 | 0.01 | 0.01 | 0.011 | 0.011 | 0.011 | 0.01 | 0.033 | 0.113 | 0.114 | 0.113 | 0.102
sensor.6_refridgerator_energy | 0.056 | 0 | 0.067 | 0.032 | 0 | 0.099 | 0.004 | 0.035 | 0.075 | 0 | 0.113 | 0.01 | 0.098 | 0.092 | 0.077 | 0 | 0.109 | 0.014 | 0.043 | 0.061 | 0.001 | 0.107 | 0 | 0.044
sensor.7_guest_bath_gfci_energy | 0.004 | 0.003 | 0.004 | 0.004 | 0.003 | 0.004 | 0.004 | 0.003 | 0.004 | 0.004 | 0.002 | 0.002 | 0.001 | 0.002 | 0.001 | 0.002 | 0.002 | 0.001 | 0.002 | 0.004 | 0.025 | 0.016 | 0.004 | 0.004
sensor.13_media_plugs_energy | 0.306 | 0.303 | 0.322 | 0.31 | 0.313 | 0.305 | 0.296 | 0.301 | 0.287 | 0.291 | 0.297 | 0.293 | 0.292 | 0.291 | 0.294 | 0.29 | 0.297 | 0.297 | 0.29 | 0.295 | 0.295 | 0.451 | 0.426 | 0.428
sensor.14_outside_gfci_energy | 0.025 | 0.003 | 0.004 | 0.004 | 0.004 | 0.003 | 0.004 | 0.004 | 0.003 | 0.004 | 0.004 | 0.004 | 0.004 | 0.003 | 0.004 | 0.004 | 0.004 | 0.003 | 0.004 | 0.003 | 0.004 | 0.004 | 0.005 | 0.007
sensor.15_dryer_energy | 0.008 | 0.001 | 0 | 0 | 0 | 0 | 0.001 | 0 | 0 | 0 | 0 | 0.001 | 0 | 0 | 0 | 0 | 0 | 0.001 | 0 | 0 | 0 | 0.144 | 0.535 | 1.127
sensor.16_dryer_energy | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0.001 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0.001 | 0 | 0 | 0.127 | 0.453 | 0.921
sensor.21_hot_water_heater_energy | 0.133 | 0.105 | 0.001 | 0 | 0.228 | 0 | 0 | 0 | 0.214 | 0 | 0 | 0 | 0.001 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0.232 | 0 | 0
sensor.22_hot_water_heater_energy | 0.134 | 0.106 | 0.001 | 0 | 0.229 | 0 | 0 | 0.001 | 0.214 | 0.001 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0.235 | 0 | 0
sensor.23_hvac_energy | 1.052 | 0.626 | 0.701 | 0.488 | 0.383 | 0 | 0 | 0 | 0 | 0 | 0.327 | 1.129 | 1.481 | 1.645 | 2.02 | 1.717 | 2.294 | 2.28 | 2.235 | 1.515 | 1.215 | 1.107 | 1.341 | 1.955
sensor.24_hvac_energy | 1.076 | 0.644 | 0.721 | 0.502 | 0.395 | 0 | 0 | 0 | 0 | 0 | 0.333 | 1.146 | 1.499 | 1.661 | 2.037 | 1.731 | 2.303 | 2.297 | 2.261 | 1.537 | 1.235 | 1.127 | 1.379 | 2.03
sensor.25_kitchen_outlets_mudroom_light_energy | 0.011 | 0.002 | 0.002 | 0.001 | 0.002 | 0.003 | 0.002 | 0.016 | 0.025 | 0.025 | 0.025 | 0.023 | 0.022 | 0.023 | 0.022 | 0.022 | 0.021 | 0.022 | 0.022 | 0.022 | 0.104 | 0.04 | 0.04 | 0.033
sensor.26_washer_energy | 0 | 0.001 | 0.001 | 0.001 | 0.001 | 0.002 | 0.001 | 0.002 | 0.001 | 0.001 | 0.002 | 0 | 0.001 | 0 | 0 | 0.001 | 0 | 0 | 0 | 0 | 0.001 | 0.041 | 0.027 | 0.005
sensor.27_spare_energy | 0.004 | 0.004 | 0.005 | 0.004 | 0.004 | 0.005 | 0.005 | 0.004 | 0.004 | 0.004 | 0.005 | 0.003 | 0.004 | 0.004 | 0.004 | 0.004 | 0.004 | 0.003 | 0.004 | 0.004 | 0.005 | 0.003 | 0.004 | 0.005
sensor.28_kitchen_sink_plugs_energy | 0.007 | 0.004 | 0.003 | 0.004 | 0.004 | 0.004 | 0.003 | 0.007 | 0.006 | 0.006 | 0.007 | 0.006 | 0.006 | 0.006 | 0.006 | 0.006 | 0.006 | 0.006 | 0.006 | 0.006 | 0.007 | 0.006 | 0.006 | 0.004

</body>

</html>

**Comment by flaviut** [comment]
The default config has total power calculated from the main current clamps, A & B:

```
  - platform: template
    lambda: return id(phase_a_power).state + id(phase_b_power).state;
    update_interval: never   # will be updated after all power sensors update via on_update trigger
    id: total_power
    device_class: power
    state_class: measurement
    unit_of_measurement: "W"
```

What do those look like? If it looks inaccurate, do they look better if you remove the `*pos` from the filters on those current clamps?

**Reply by FedgarLurch** [reply]
I don't know how to check the raw numbers for each phase. My wires are too large and in the wrong place to put an amp clamp directly on. I have a parallel main coming in from the utility meter, so I have a clamp on two of the wires, and the multiplier set to 2. 

I removed the *pos and I will let it ride for the rest of the night and see after 8am when I get the report from my utility what the new numbers look like.

Thanks!
